### PR TITLE
Implement PyOpenCLArrayContext.ravel

### PR DESCRIPTION
**Pull request by kaushikcfd**

### CODE REVIEW

File: arraycontext/impl/pyopencl.py [hunk 1]
```diff
@@ -141,6 +142,25 @@ def concatenate(self, arrays, axis=0):
             self._array_context.allocator
         )
 
+    def ravel(self, a, order="C"):
+        def _rec_ravel(a):
+            if order == "K":
```
Comment (alexfikl) [inline arraycontext/impl/pyopencl.py:147]: What about `order == "A"`? Raise a `NotImplementedError`?

https://numpy.org/doc/stable/reference/generated/numpy.ravel.html#numpy.ravel

Comment (alexfikl) [reply arraycontext/impl/pyopencl.py:147]: Actually, from the docs it seems like `"A"` is the one that does Fortran order or C order depending on which one is contiguous. Am I misreading that?

Comment (kaushikcfd) [reply arraycontext/impl/pyopencl.py:147]: Yeah this should really be "A" and not implemented for "K".

Comment (kaushikcfd) [reply arraycontext/impl/pyopencl.py]: Thanks, see 0e11990.

**Pull request merged**